build.gradle

```diff
@@ -104,7 +132,309 @@ subprojects { project ->
+    mustRunAfter prepareDocResources
+    from('src/main/docs/resources')
+    into docResourcesDir
+}
+
+configurations {
+    documentation
+}
+
+dependencies {
+    if (configurations.find { it.name == 'documentation' }) {
+        documentation 'info.picocli:picocli:3.8.0'
+        documentation "org.fusesource.jansi:jansi:1.18"
+        documentation "jline:jline:2.14.6"
+        documentation "com.github.javaparser:javaparser-core:3.15.14"
+        documentation "org.apache.groovy:groovy:${groovyVersion}"
+        documentation "org.apache.groovy:groovy-templates:${groovyVersion}"
+        documentation "org.apache.groovy:groovy-ant:$groovyVersion"
+        documentation "org.apache.groovy:groovy-cli-picocli:$groovyVersion"
+    }
+}
+
+def cleanTask = project.tasks.findByName("clean")
+if (cleanTask == null) {
+    task clean(type: Delete) {
+        delete(buildDir)
+    }
+} else {
+    cleanTask.doLast {
+        ant.delete(dir: "build/docs")
+    }
+}
+
+tasks.withType(Groovydoc) {
+    group = DOCUMENTATION_GROUP
+    docTitle = "${project.title} - ${project.version}"
+    destinationDir = project.file("build/docs/api")
+    def files = []
+    project.rootProject.subprojects
+            .findAll { !it.name != 'docs' && !it.name.startsWith('examples') }
+            .each { subproject ->
+                if (subproject.file('src/main/groovy').exists()) {
+                    files += subproject.files("src/main/groovy")
+                }
+            }
+    if (project.file('src/main/groovy').exists()) {
+        files += project.files("src/main/groovy")
+    }
+    source = files
+    classpath += configurations.documentation
+}
+
+tasks.withType(org.gradle.api.tasks.javadoc.Groovydoc) {
+    configure {
+        access = GroovydocAccess.PRIVATE
+        processScripts = false
+        includeMainForScripts = false
+        includeAuthor = true
+        groovyClasspath = configurations.documentation
+    }
+}
+
+task publishGuide(type: grails.doc.gradle.PublishGuide) {
+    group = DOCUMENTATION_GROUP
+    description = 'Generate Guide'
+    dependsOn copyLocalDocResources
+
+    targetDir = project.file("${buildDir}/docs")
+    sourceRepo = "https://github.com/${githubSlug}/edit/${githubBranch}/src/main/docs"
+    sourceDir = new File(projectDir, "src/main/docs")
+    propertiesFiles = [new File(rootProject.projectDir, "gradle.properties")]
+    asciidoc = true
+    resourcesDir = project.file(docResourcesDir)
+    properties = ['safe'     : 'UNSAFE',
+                  'version'  : project.version,
+                  'subtitle' : project.projectDesc,
+                  'api'      : '../api',
+                  'sourceDir': rootProject.projectDir.absolutePath,
+                  'sourcedir': rootProject.projectDir.absolutePath,
+                  'javaee'   : 'https://docs.oracle.com/javaee/7/api/',
+                  'javase'   : 'https://docs.oracle.com/javase/7/docs/api/',
```

Comment:
These should probably be updated at some point

Reply:
Updated